public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,347 @@
+      savedAugmentDetectorsSet.has(detector.id)
+    );
+    return detectorsToDisplay;
+  };
+
+  const onUnlinkDetector = async () => {
+    setIsLoadingFinalDetectors(true);
+    await savedObjectLoader.findAll().then(async (resp: any) => {
+      if (resp != undefined) {
+        const savedAugmentObjects: ISavedAugmentVis[] = get(resp, 'hits', []);
+        // gets all the saved object for this visualization
+        const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+          savedAugmentObjects.filter(
+            (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+          );
+
+        // find saved Augment object matching detector we want to unlink
+        // There should only be one detector and vis pairing
+        const savedAugmentToUnlink = savedAugmentForThisVisualization.find(
+          (savedObject) =>
+            get(savedObject, 'pluginResourceId', '') === detectorToUnlink.id
+        );
+        const savedObjectToUnlinkId = get(savedAugmentToUnlink, 'id', '');
+        await savedObjectLoader
+          .delete(savedObjectToUnlinkId)
+          .then(async (resp: any) => {
+            core.notifications.toasts.addSuccess({
+              title: `Association removed between the ${detectorToUnlink.name}
+              and the ${embeddableTitle} visualization`,
+              text: "The detector's anomalies do not appear on the visualization. Refresh your dashboard to update the visualization",
+            });
+          })
+          .catch((error) => {
+            core.notifications.toasts.addDanger(
+              prettifyErrorMessage(
+                `Error unlinking selected detector: ${error}`
+              )
+            );
+          })
+          .finally(() => {
+            getDetectors();
+          });
+      }
+    });
+  };
+
+  const getUnlinkConfirmModal = () => {
+    if (confirmModalState.isOpen) {
+      return (
+        <ConfirmUnlinkDetectorModal
+          detector={confirmModalState.affectedDetector}
+          onUnlinkDetector={onUnlinkDetector}
+          onHide={handleHideModal}
+          onConfirm={handleConfirmModal}
+          isListLoading={isLoading}
+        />
+      );
+    }
+  };
+
+  const handleHideModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isOpen: false,
+    });
+  };
+
+  const handleConfirmModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isRequestingToClose: true,
+    });
+  };
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  // TODO: this part is incomplete because it is pending on a different PR that will have all the associate existing changes
+  const onAssociateExistingDetector = async () => {
+    console.log('inside create anomaly detector');
+  };
+
+  const handleUnlinkDetectorAction = (detector: DetectorListItem) => {
+    setDetectorToUnlink(detector);
+    if (!isEmpty(detector)) {
```

Comment:
When would the stored detector list item be empty?

Reply:
I actually don't think it would be possible on a second thought here since its coming directly from the list of detectors. I think originally I was thinking that there would be some lag if detector was already unlinked but I guess it wouldn't be in this operation cause its coming from the stored list and could maybe happen on the crud operations where i do have error checking

Reply:
Cool - let's go ahead and remove it